modules/NEON/variables.tf

```diff
@@ -0,0 +1,306 @@
+  type        = bool
+  default     = true
+}
+
+variable "container_image_repository" {
+  description = "Container Image Repo, only set if creating container image notebook instance by setting `create_container_image` variable to true. {{UIMeta group=3 order=4 }}"
+  type        = string
+  default     = ""
+}
+
+variable "container_image_tag" {
+  description = "Container Image Tag, only set if creating container image notebook instance by setting `create_container_image` variable to true. {{UIMeta group=3 order=5 }}"
+  type        = string
+  default     = "latest"
+}
+
+variable "deployment_id" {
+  description = "Adds a suffix of 4 random characters to the `project_id`."
+  type        = string
+  default     = null
+}
+
+variable "enable_gpu_driver" {
+  description = "Install GPU driver on the instance. {{UIMeta group=3 order=9 }}"
+  type        = bool
+  default     = false
+}
+
+variable "enable_services" {
+  description = "Enable the necessary APIs on the project.  When using an existing project, this can be set to false. {{UIMeta group=1 order=3 }}"
+  type        = bool
+  default     = true
+}
+
+variable "folder_id" {
+  description = "Folder ID where the project should be created. It can be skipped if already setting organization_id. Leave blank if the project should be created directly underneath the Organization node. {{UIMeta group=0 order=2 updatesafe }}"
+  type        = string
+  default     = ""
+}
+
+variable "gpu_accelerator_type" {
+  description = "Type of GPU you would like to spin up. {{UIMeta group=3 order=10 }}"
+  type        = string
+  default     = ""
+}
+
+variable "gpu_accelerator_core_count" {
+  description = "Number of of GPU core count. {{UIMeta group=3 order=11 }}"
+  type        = number
+  default     = 0
+}
+
+variable "image_family" {
+  description = "Image of the AI notebook. {{UIMeta group=3 order=13 }}"
+  type        = string
+  default     = "tf-latest-cpu"
+}
+
+variable "image_project" {
+  description = "Google Cloud project where the image is hosted. {{UIMeta group=3 order=12 }}"
+  type        = string
+  default     = "deeplearning-platform-release"
+}
+
+variable "ip_cidr_range" {
+  description = "Unique IP CIDR Range for AI Notebooks subnet. {{UIMeta group=2 order=5 }}"
+  type        = string
+  default     = "10.142.190.0/24"
+}
+
+variable "machine_type" {
+  description = "Type of VM you would like to spin up. {{UIMeta group=3 order=6 }}"
+  type        = string
+  default     = "n1-standard-4"
+}
+
+variable "network_name" {
+  description = "Name of the network to be created. {{UIMeta group=2 order=2 }}"
+  type        = string
+  default     = "neon-network"
+}
+
+variable "notebook_count" {
+  description = "Number of Vertex AI Notebooks requested. {{UIMeta group=3 order=2 updatesafe }}"
```

Comment:
This could be destructive.

Reply:
Added a note in the variable description.